### ISSUE TITLE
Build fails on Windows 10

### ISSUE THREAD

**Issue by ynnirc** [telegramdesktop/tdesktop#24241]
### Steps to reproduce

Follow https://github.com/telegramdesktop/tdesktop/blob/dev/docs/building-win-x64.md

### Expected behaviour

It builds

### Actual behaviour

Here is the output of `configure.bat`:
```
Warn about uninitialized values.
-- Building for: Visual Studio 17 2022
Version: 3.6.1
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19044.
-- The C compiler identification is MSVC 19.30.30706.0
-- The CXX compiler identification is MSVC 19.30.30706.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.30.30705/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.30.30705/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Python: C:/Users/crinny/AppData/Local/Programs/Python/Python310/python.exe (found version "3.10.1") found components: Interpreter
CMake Warning at cmake/validate_d3d_compiler.cmake:11 (message):
  Bad "d3dcompiler_47.dll" version: 10.0.19041.685

Call Stack (most recent call first):
  cmake/validate_d3d_compiler.cmake:59 (validate_d3d_error)
  cmake/win_directx_helper/CMakeLists.txt:16 (validate_d3d_compiler)


Resolved NuGet executable: C:/Users/crinny/TBuild/ThirdParty/NuGet/nuget.exe
Feeds used:

Argument cannot be null or empty
Parameter name: primarySources
CMake Error at cmake/nuget.cmake:86 (execute_process):
  execute_process error getting child return code: The system cannot find the
  file specified
Call Stack (most recent call first):
  Telegram/lib_base/CMakeLists.txt:262 (nuget_add_winrt)


-- Configuring incomplete, errors occurred!
See also "C:/Users/crinny/TBuild/tdesktop/out/CMakeFiles/CMakeOutput.log".
```

### Operating system

Windows 10 x64, Strawberry Perl 5.32.1.1, NASM 2.15.05, Yasm 1.3.0, MSYS2 20220319, jom 1.1.3, Python 3.9.11, CMake 3.23.0, Ninja 1.7.2, Visual Studio 2022

### Version of Telegram Desktop

dev commit bf3fbd41412f36bb959d5050ed1a30710072b54a

### Installation source

Static binary from official website

**Comment by cxyLi** [comment]
i  hava  meet  the  same problem！

**Comment by c0re100** [comment]
Make sure you've installed Visual Studio 2022 with Windows 11 SDK (10.0.22000.0)

**Comment by cxyLi** [comment]
D:\Telegram\tdesktop\Telegram>configure.bat x64 -D TDESKTOP_API_ID=17349  -D TDESKTOP_API_HASH=344583e45741c457fe1862106095a5eb  -D DESKTOP_APP_USE_PACKAGED=OFF -D DESKTOP_APP_DISABLE_CRASH_REPORTS=OFF
Warn about uninitialized values.
Version: 3.6.1
-- Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.19042.
CMake Error at Telegram/CMakeLists.txt:12 (add_subdirectory):
  The source directory

    D:/Telegram/tdesktop/Telegram/lib_base

  does not contain a CMakeLists.txt file.


CMake Error at Telegram/CMakeLists.txt:14 (add_subdirectory):
  The source directory

    D:/Telegram/tdesktop/Telegram/lib_tl

  does not contain a CMakeLists.txt file.


CMake Error at Telegram/CMakeLists.txt:19 (add_subdirectory):
  The source directory

    D:/Telegram/tdesktop/Telegram/lib_lottie

  does not contain a CMakeLists.txt file.


CMake Error at Telegram/CMakeLists.txt:24 (add_subdirectory):
  The source directory

    D:/Telegram/tdesktop/Telegram/lib_webrtc

  does not contain a CMakeLists.txt file.


Resolved NuGet executable: D:/Telegram/ThirdParty/NuGet/nuget.exe
Feeds used:
  C:\Users\Administrator\.nuget\packages\
  https://api.nuget.org/v3/index.json
  C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\

已安装程序包“Microsoft.Web.WebView2.1.0.864.35”。
Feeds used:
  C:\Users\Administrator\.nuget\packages\
  https://api.nuget.org/v3/index.json
  C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\

已安装程序包“Microsoft.Windows.CppWinRT.2.0.210505.3”。
Generating new WinRT headers.
CMake Error at Telegram/CMakeLists.txt:26 (add_subdirectory):
  The source directory

    D:/Telegram/tdesktop/Telegram/codegen

  does not contain a CMakeLists.txt file.


Using existing WinRT headers.
-- Configuring incomplete, errors occurred!
See also "D:/Telegram/tdesktop/out/CMakeFiles/CMakeOutput.log".

**Comment by cxyLi** [comment]
thanks very much !   it  seems not work!

**Comment by ynnirc** [comment]
> Make sure you've installed Visual Studio 2022 with Windows 11 SDK (10.0.22000.0)

Still couldn't get it to work with that SDK installed.

**Comment by danarrib** [comment]
First I have to say... boy... Telegram build process has TOO MANY dependencies. It's bonkers. I spent almost 4 hours just to build the environment.

Ok, about the problem... The instructions provided by [the referenced document](https://github.com/telegramdesktop/tdesktop/blob/dev/docs/building-win-x64.md) are a bit outdated and/or incomplete.

![image](https://user-images.githubusercontent.com/17026744/163734049-4f02018e-a05e-4b5f-8118-6e45c4f2dc57.png)

This command runs a batch file that tries to build every single dependency. But the 11th, 13th and 16th dependencies (ffmpeg, brakepad and tg_owt) fails to build.

The last one fails because ffmpeg failed before. So you fix ffmpeg and tg_owt will be fixed too. Unfortunately, I could not fix the ffmpeg build. It would be nice if someone can give some advice on that, details are at the end of this comment.

**To solve the brakepad build problem**, just have to create a `GYP_MSVS_OVERRIDE_PATH` system environment variable, pointing to your Visual Studio 2022 folder. In my case:

```
set GYP_MSVS_OVERRIDE_PATH=D:\Program Files\Microsoft Visual Studio\2022\Community
```

_Tip: don't forget to close the command prompt windows and open it again after changing system variables. They only load the values once, during startup._

Also, I had to **copy all files and subdirectories** from `[VS Directory]\VC\Auxiliary\Build` to `[VS directory]\VC` (two folders above), because brakepad somehow requires C compiler files on the VC folder and not inside the "Auxiliary\Build" subfolders. Maybe Microsoft changed this behavior on the 2022 version and nobody updated the build scripts for it.

**Now, about ffmpeg build problem:**

I found some information online about creating this two system environment variables:
```
set SET_FULL_PATH=1
set MSYS2_BIN="D:\TBuild\ThirdParty\msys64\usr\bin\bash.exe"
```
And also adding to the `PATH` system variable the path to the **MSVC** binaries and to **msys64**.
```
D:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.30.30705\bin\Hostx64\x64
D:\TBuild\ThirdParty\msys64
```

Doing that in fact changed the error message. But it's still getting an error when building.
At some point, the build script builds a "test.exe" program. And right after that, it tries to run this program but it returns an error.

![image](https://user-images.githubusercontent.com/17026744/163734743-ad4f8317-c3e6-4f2f-8ab5-766a92bbd840.png)
```
cl.exe is unable to create an executable file.
If cl.exe is a cross-compiler, use the --enable-cross-compile option.
Only do this if you know what cross compiling means.
C compiler test failed.
```

The last lines of `ffmppeg/ffbuild/config.log` are:
```
test_ld cc
test_cc
BEGIN ./ffconf.lz3ohUfm/test.c
    1	int main(void){ return 0; }
END ./ffconf.lz3ohUfm/test.c
cl.exe -nologo -DCONFIG_SAFE_BITSTREAM_READER=1 -c -Fo./ffconf.lz3ohUfm/test.o ./ffconf.lz3ohUfm/test.c
test.c
./compat/windows/mslink -nologo -out:./ffconf.lz3ohUfm/test.exe ./ffconf.lz3ohUfm/test.o
./configure: line 1463: ./ffconf.lz3ohUfm/test.exe: Permission denied
C compiler test failed.
```

So I tried to find this **ffconf.lz3ohUfm** directory to see this "test.o" and "test.c" files, but the directory actually doesn't exists... In fact, I noticed that this directory is created while build ffmpeg, and deleted just before the build script ends. And also, this characters after "ffconf." are random, so every time you run the build script, the directory name changes.

Ok, so far what I know is: It really writes a "test.exe" program, but the program does not run. Windows gives a "permission denied" error. And the program is deleted just after that, so I could not even save it to see what happens.

Yes, Terminal is running on elevated permissions (Run as Administrator).

I tried to find other useful information about that on ffmpeg build problems, but had no success.
